packages/async-flow/src/replay-membrane.js

```diff
@@ -196,9 +202,61 @@ export const makeReplayMembrane = (
       default: {
         // @ts-expect-error TS correctly knows this case would be outside
         // the type. But that's what we want to check.
-        throw Fail`unexpected outcome kind ${q(outcome.kind)}`;
+        throw PanicTag`unexpected outcome kind ${q(outcome.kind)}`;
+      }
+    }
+  };
+
+  // //////////////// Eventual Send ////////////////////////////////////////////
+
+  const guestHandler = harden({
+    applyMethod(guestTarget, optVerb, guestArgs, guestReturnedP) {
+      if (optVerb === undefined) {
+        throw PanicTag`guest eventual call not yet supported: ${guestTarget}(${b(guestArgs)}) -> ${b(guestReturnedP)}`;
+      } else {
+        throw PanicTag`guest eventual send not yet supported: ${guestTarget}.${b(optVerb)}(${b(guestArgs)}) -> ${b(guestReturnedP)}`;
       }
+    },
+    applyFunction(guestTarget, guestArgs, guestReturnedP) {
+      return guestHandler.applyMethod(
+        guestTarget,
+        undefined,
+        guestArgs,
+        guestReturnedP,
+      );
+    },
+    get(guestTarget, prop, guestReturnedP) {
+      throw PanicTag`guest eventual get not yet supported: ${guestTarget}.${b(prop)} -> ${b(guestReturnedP)}`;
+    },
+  });
+
+  const makeGuestPresence = (iface, methodEntries) => {
+    let guestPresence;
+    void new HandledPromise((_res, _rej, resolveWithPresence) => {
+      guestPresence = resolveWithPresence(guestHandler);
+    }); // no unfulfilledHandler
+    if (typeof guestPresence !== 'object') {
+      throw Fail`presence expected to be object ${guestPresence}`;
     }
+    assign(guestPresence, fromEntries(methodEntries));
+    const result = Remotable(iface, undefined, guestPresence);
+    result === guestPresence ||
+      Fail`Remotable expected to make presence in place: ${guestPresence} vs ${result}`;
+    return result;
+  };
```

Comment:
From my understanding of HandledPromise this seems correct, but I really would be more confident if @michaelfig could double check, as this API is fairly convoluted.

Reply:
Confirmed.  These are the correct contortions given the current HandledPromise API.